src/Dns/Dns/Zones/NewAzureDnsZone.cs

```diff
@@ -102,7 +115,48 @@ public override void ExecuteCmdlet()
                         ? string.Format(ProjectResources.Success_NewPrivateZone, this.Name, this.ResourceGroupName)
                         : string.Format(ProjectResources.Success_NewZone, this.Name, this.ResourceGroupName));
                     this.WriteObject(result);
+
+                    if (!string.IsNullOrEmpty(this.ParentZoneName) && this.Name.EndsWith(this.ParentZoneName))
```

Comment:
@ShwethaRaghupathi I think that we should make these series of calls before we call `this.WriteObject(result)` above -- this will ensure that if there were any problems with adding the delegation, the cmdlet will fail before an object is written to the output stream

Reply:
The requirement is that zone creation and delegation need not be atomic. Zone creation if successful and delegation does not go through, indicate success of zone creation following which indicate that delegation could not be added. Hence not moving it to the end.

Reply:
@ShwethaRaghupathi this pattern can be confusing to users as there is a scenario where the cmdlet will return the zone that they created, but could also throw an exception after that if the delegation was unsuccessful -- this is the issue with having two completely separate operations put into a single cmdlet. I think what would be best is if we caught any errors when trying to do the delegation and wrote a warning saying that the delegation was unsuccessful rather than throwing an exception.

Reply:
Yes that makes sense to indicate via warning msg. Updated the PR with this change